Source/Csla.Blazor/State/StateManager.cs

```diff
@@ -74,7 +74,7 @@ private async Task<Session> WaitForState(TimeSpan timeout)
         var tcs = new TaskCompletionSource();
         session.PropertyChanged += (s, e) =>
         {
-          if (e.PropertyName == "IsCheckedOut" && !session.IsCheckedOut)
+          if (e.PropertyName == "IsCheckedOut" && !session.IsCheckedOut && !tcs.Task.IsCompleted)
```

Comment:
I think instead of this you should propery unregister the delegate.
A local function with a try{}finally{} is the right way. I'm not sure if this otherwise will lead to a memory leak due to the closure.